File: go/p3/p3_v1.go
package main

import (
	"fmt"
	"time"
)

func problem3() int {
	value := 600851475143
	largestPrimeFactor := 0
	nextPossibleFactor := 2

	for value != 1 {
		if value%nextPossibleFactor == 0 {
			largestPrimeFactor = nextPossibleFactor
			value /= nextPossibleFactor
		} else {
			nextPossibleFactor += 1 + nextPossibleFactor%2
		}
	}

	return largestPrimeFactor
}

func main() {
	start := time.Now()
	fmt.Println("Result:", problem3())
	elapsed := time.Since(start)
	fmt.Printf("Execution time: %.6f seconds\n", elapsed.Seconds())
}

File: python/p3/p3_v1.py
def problem3():
    value = 600851475143

    largest_prime_number = 0

    number = 2
    while value != 1:
        if value % number == 0:
            value /= number
            largest_prime_number = number
        else:
            number += 1 if number == 2 else 2
    return largest_prime_number

if __name__ == "__main__":
    import time
    
    start_time = time.time()
    result = problem3()
    end_time = time.time()

    print(f"Result: {result}")
    print(f"Execution time: {end_time - start_time:.6f} seconds")